File: 2490-circular-sentence/2490-circular-sentence.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        if sentence[0]!=sentence[-1]:
            return False
        
        words_list = sentence.split(" ")
        for i in range(len(words_list) - 1):
            if words_list[i][-1] != words_list[i + 1][0]:
                return False
        
        return True

File: 2490-circular-sentence/2490-circular-sentence.cpp
class Solution {
public:
    bool isCircularSentence(string sentence) {
        if (sentence.front() != sentence.back())
            return false;
        
        // Split the sentence by spaces and store each word in a vector
        std::istringstream stream(sentence);
        std::vector<std::string> words;
        std::string word;
        while (stream >> word) {
            words.push_back(word);
        }
        
        // Iterate through each word and check if the last character of the current word
        // matches the first character of the next word
        for (size_t i = 0; i < words.size() - 1; ++i) {
            if (words[i].back() != words[i + 1].front()) {
                return false;
            }
        }
        
        return true;
    }
};